### PR TITLE
PCA demo notebook

### PR DESCRIPTION
**Pull request by bkmartinjr**
Changes:
* added the X sparse iterator to the experimental package. As part of this, moved location of _eager_iter.py
* added start at a PCA demo notebook

### CODE REVIEW

File: api/python/cellxgene_census/pyproject.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@ dependencies= [
     "requests",
     "typing_extensions",
     "s3fs>=2021.06.1",
-    "scipy",
+    "scipy<=1.10.1",  # for compat with pyarrow
```
Comment (ebezzi) [inline api/python/cellxgene_census/pyproject.toml:41]: Needs reconciliation with the other PR.

File: api/python/notebooks/experimental/pca.ipynb [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+     "text": [
+      "The \"latest\" release is currently 2023-06-20. Specify 'census_version=\"2023-06-20\"' in future calls to open_soma() to ensure data consistency.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "36564 cells selected\n",
+      "Begining HVG calculation\n",
```
Comment (ebezzi) [inline api/python/notebooks/experimental/pca.ipynb:63]:
```suggestion
      "Beginning HVG calculation\n",
```

File: api/python/notebooks/experimental/pca.ipynb [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+   "source": [
+    "with cellxgene_census.open_soma(census_version=census_version) as census:\n",
+    "    exp = census[\"census_data\"][experiment_name]\n",
+    "\n",
+    "    with exp.axis_query(\n",
+    "        measurement_name=\"RNA\",\n",
+    "        obs_query=soma.AxisQuery(value_filter=obs_value_filter),\n",
+    "    ) as query:\n",
+    "        print(f\"{query.n_obs} cells selected\")\n",
+    "        print(\"Begining HVG calculation\")\n",
```
Comment (ebezzi) [inline api/python/notebooks/experimental/pca.ipynb:483]:
```suggestion
    "        print(\"Beginning HVG calculation\")\n",
```

**Review by ebezzi — APPROVED** [review]
LGTM 🚀

**Comment by pablo-gar** [comment]
Answering some of your questions from your email:

> when doing the training, on the raw counts, are there any normalizations and/or transformations that should be done? 
Intuitively, I suspect that there needs to be some type of normalization across datasets, but I'd like advice on what it should be.

Yes, the workflow usually goes:

1. Filter genes out, expressed in < N cells (N = 50?, needs parametrization)
2. Row-based (obs) normalization of sequencing depth: `X[i,] = X[i,] / sum(X[i,])`, scanpy equivalent `pp.normalize_total`
3. For Smart-seq data further normalize by gene-length: `X[i,] = X[i,] / gene_lengths[i,])`
4. Log transform values
5. Highly variable genes selection
6.  [OPTIONAL; NEEDS EXPLORATION] regress out dataset-batch effects with linear regression, scanpy equivalent `pp.regress_out` 
7. Column-based (var) centering and scaling (z-scores), scanpy equivalent `pp.scale`
8. Cap values to a threshold, usually 10 (no more than 10 standard deviations away from average)

I think you tackled already 5

**Review by atolopko-czi — APPROVED** [review]
Notebook code LGTM.

How long does it take to run as configured? Want to ensure it's easy for anyone to re-run all of the notebooks for LTS releases. The current [README](https://github.com/chanzuckerberg/cellxgene-census/blob/8b75ae5bb36b9f18b03fac8a4d45262c6d218c78/api/python/notebooks/README.md#L1) suggests using a m6i.8xlarge, e.g.

Optionally, could factor out these parameters to the first code cell for tuning awareness/ease:
* `training_chunk_size`
* `stride=2**17`

The all-in-one code block works for me, but @pablo-gar may have thoughts on whether we want more "commentary", which may force this to be broken into multiple cells w/accompanying explanation. I'm not advocating for that, since the code is straightforward to me.

**Comment by bkmartinjr** [comment]
I"m not even sure @pablo-gar wants this landed or deleted, as it may not be all that useful as written.  Will add to weekly agenda.

**Comment by bkmartinjr** [comment]
@pablo-gar - ping.  What do you want to do with this?  (change, land, delete?)  TY!

**Pull request merged**